test/distributed/fsdp/test_fsdp_optim_state.py

```diff
@@ -243,23 +244,32 @@ def _init_nested_model(
                 {"params": model.param_group0()},
                 {"params": model.param_group1(), "weight_decay": 0.9}
             ]
+        # Use a reversed parameter order for the optimizer input on odd ranks
+        if use_diff_optim_inputs and self.rank % 2 == 1:
```

Comment:
Do we get a similar error as we've seen internally with this change, without fixing the optim_state_dict logic?